packages/ckeditor5-table/src/tableclipboard.js

```diff
@@ -74,4 +80,163 @@ export default class TableClipboard extends Plugin {
+		const { first: firstRowOfSelection, last: lastRowOfSelection } = getRowIndexes( selectedTableCells );
+
+		const selectionHeight = lastRowOfSelection - firstRowOfSelection + 1;
+		const selectionWidth = lastColumnOfSelection - firstColumnOfSelection + 1;
+
+		const insertHeight = tableUtils.getRows( insertedTable );
+		const insertWidth = tableUtils.getColumns( insertedTable );
+
+		// The if below is temporal and will be removed when handling this case.
+		// See: https://github.com/ckeditor/ckeditor5/issues/6769.
+		if ( selectionHeight > insertHeight || selectionWidth > insertWidth ) {
+			// @if CK_DEBUG // console.log( 'NOT IMPLEMENTED YET: Pasted table is smaller than selection area.' );
+
+			return;
+		}
+
+		const model = this.editor.model;
+
+		model.change( writer => {
+			// Crop pasted table if it extends selection area.
+			if ( selectionHeight < insertHeight || selectionWidth < insertWidth ) {
+				insertedTable = cropTableToDimensions( insertedTable, 0, 0, selectionHeight - 1, selectionWidth - 1, tableUtils, writer );
+			}
+
+			// Stores cells anchors map of inserted table cell as '"row"x"column"' index.
+			const insertionMap = new Map();
+
+			for ( const { column, row, cell } of new TableWalker( insertedTable ) ) {
+				insertionMap.set( `${ row }x${ column }`, cell );
+			}
+
+			// Content table to which we insert a table.
+			const contentTable = findAncestor( 'table', selectedTableCells[ 0 ] );
+
+			// Selection must be set to pasted cells (some might be removed or new created).
+			const cellsToSelect = [];
+
+			// Store previous cell in order to insert a new table cells after it if required.
+			let previousCellInRow;
+
+			const tableMap = [ ...new TableWalker( contentTable, {
+				startRow: firstRowOfSelection,
+				endRow: lastRowOfSelection,
+				includeSpanned: true
+			} ) ];
+
+			for ( const { column, row, cell, isSpanned } of tableMap ) {
+				if ( column === 0 ) {
+					previousCellInRow = null;
+				}
+
+				// Could use startColumn, endColumn. See: https://github.com/ckeditor/ckeditor5/issues/6785.
+				if ( column < firstColumnOfSelection || column > lastColumnOfSelection ) {
+					previousCellInRow = cell;
+
+					continue;
+				}
+
+				// Map current table location to inserted table location.
+				const cellLocationToInsert = `${ row - firstRowOfSelection }x${ column - firstColumnOfSelection }`;
+				const cellToInsert = insertionMap.get( cellLocationToInsert );
+
+				// There is no cell to insert (might be spanned by other cell in a pasted table) so...
+				if ( !cellToInsert ) {
+					// ...if the cell is anchored in current location (not-spanned slot) then remove that cell from content table...
+					if ( !isSpanned ) {
+						writer.remove( writer.createRangeOn( cell ) );
+					}
+
+					// ...and advance to next content table slot.
+					continue;
+				}
+
+				let targetCell = cell;
```

Comment:
This seems not used before overridden later.

Reply:
Might be refactor leftover.